### ISSUE TITLE
Gradual Infill steps improvement

### ISSUE THREAD

**Issue by AHS87** [Ultimaker/Cura#9374]
### Is your feature request related to a problem?

When using the infill steps cura draws some lines on open space

### Describe the solution you'd like

To add an option that forces the new layer to begin on a place where there is material on the previous layer

### Describe alternatives you've considered

A solution could be to add an option that forces the new step to travel the full length of the previous step for a certain amount of layers.

### Affected users and/or printers

Everyone printing something with an irregular top layer who wants to save some material in the process

This is how it is working
![Steps](https://user-images.githubusercontent.com/80118887/110147987-e3ccf980-7da1-11eb-827e-ef4f6dc482f6.png)

This is what I propose
![Steps_full_length](https://user-images.githubusercontent.com/80118887/110148018-ecbdcb00-7da1-11eb-858d-c8e8e4388eac.png)

**Comment by fvrmr** [comment]
Hi @AHS87 thank you for your feature request.
I will discuss it with the team. Keep you posted.

**Comment by fvrmr** [comment]
Hi @AHS87 try increasing your step size. 
And if that is not helping, what is the impact on your print? 
Because if you look at layers above this layer you will see that it will connect.

**Comment by AHS87** [comment]
Depending on the complexity of the top layer sometimes the steps take place on empty space, I think having some layers traveling the full length of the previous step will allow for complex surfaces and shorter steps. Sometimes it can be solved by taking long steps but this uses more material.

**Comment by AHS87** [comment]
This is only an idea to improve this amazing feature allowing for prints with less material when prototyping or for prints that don't need strong infill, I hope it is taken into account for a future version.

**Comment by Ghostkeeper** [comment]
In my mind, this is the same sort of problem as https://github.com/Ultimaker/Cura/issues/8983 . Improvements can certainly be made for certain infill patterns. But the same improvements don't apply to concentric infill, Cross infill, Gyroid, etc.

**Comment by AHS87** [comment]
You are right, this will help only with some patterns. The idea is to be able to do prototypes saving material and time by using extremely low infill and allowing for complex top layers with a relative good quality.

**Comment by fvrmr** [comment]
I had a small discussion with a colleague about this issue because we didn't reach a conclusion and got a new issue related to this, but we decided to defer this issue.
The reason why is that it would build up eventually to be supported and like @Ghostkeeper mentioned above it is made for certain infill patterns.

**Comment by dsvilko** [comment]
It's very unfortunate that this is closed. Printing in mid-air is never good and can often lead to problems. I really believe the infill algorithm should be modified such that it can only start and stop the printing of a line over a previously printed line of the previous layer. For objects with complex geometry gradual infill steps currently provide very little or no roof support making the option useless for anything other than simple, flat roofs.

**Comment by fvrmr** [comment]
@dsvilko it is not closed, we only made the decision that we don't have the time and priority to add this ourselves now. Everybody else is always welcome to make a pull request for this.

**Comment by Ghostkeeper** [comment]
Or we could see new opportunities in this later if Ultimaker's priorities shift or if new reasons arise why this needs to be implemented. Repeating the old arguments won't make much of a difference, but if you can argue that there is a good reason why this should get higher priority, we do listen.

Currently, our experience is that it's enough for an infill line to attach to the wall on one side. The line will stick there and provide a bit more support for the infill line on the next layer, so that it can provide more support on the next layer and so on. In practice the infill is fully supported after a few layers. Attaching the infill lines on both sides could reduce the number of layers it takes for this to be properly supported, at the cost of using some extra material on those layers. We estimate that it'll save very little material and time in the end.
Other disadvantages of this approach are:
* Increased slicing time with grid, lines, triangles, tri-hexagon, cubic, octet and quarter cubic patterns to figure out where the next line intersects.
* Can't work with Concentric.
* Can't work with zigzag?
* Can't work with lines?
* It's unsure how this would work with the Cross, Cross 3D and Gyroid infill patterns, but we could be looking at simply generating the infill pattern at two densities and doing line-by-line intersection which is very expensive.
* The increased slicing time could expand beyond the time saved in printing, or at least beyond equal time value assuming operator time is 10 times as valuable as printer time.

Those are the reasons why we don't consider this to be worth our time investment at the moment, so we didn't add it to our priority list. If anything happens to change those reasons, it can certainly be reconsidered.

**Comment by SxeyChick** [comment]
I am affirming that this issue does affect simple prints to full failure with almost every single infill. 

Here is a simple box with an embossed word on in. A simple a common thing to print. 

You can see that for every single infill (other than cubesub, cross, and cross3d) the gradual infill under the word is either fully unsupported, or 90% unsupported, both would result in full failed of the word to print. 

This issue needs to be fixed. 

My suggestion! The gradual infill could act like lightning infill off of the other infill or the walls of the print if it’s not supported on both ends (or at all).

![162D31F1-28B9-4AE1-B120-8C969DE8C040](https://user-images.githubusercontent.com/115029650/193878378-4bd85158-c97f-43c6-a4a6-7b0ec2942727.jpeg)
![605F48BF-2BD1-4849-A5FE-8E6F2938FF03](https://user-images.githubusercontent.com/115029650/193878385-c72636c2-a90e-445e-a80a-b95ab974ad20.jpeg)

![4B68CCAC-5351-41C2-AE63-8B8F79C4A9B0](https://user-images.githubusercontent.com/115029650/193878230-5f65b506-0c0a-40a0-b47c-81f5f065d110.jpeg)
![9456B86C-6F5E-4E60-8ABD-BF7D86601BDB](https://user-images.githubusercontent.com/115029650/193878254-a49ffb22-1ebf-40b6-928b-dcddd0f1ac29.jpeg)
![564225C6-D1FA-487C-9A07-C6A487D28756](https://user-images.githubusercontent.com/115029650/193878263-5b864025-0841-4aeb-b486-9639286b15bd.jpeg)
![33DCF320-FE34-4D90-B50F-C86A974B8388](https://user-images.githubusercontent.com/115029650/193878273-3fa6d414-3fc0-4842-961d-fd6eaeb5a5c4.jpeg)
![CA5CFFCD-457B-4604-A502-08C1F767E72B](https://user-images.githubusercontent.com/115029650/193878290-12fc3cb2-c38f-499a-8632-4980f7b0fd7a.jpeg)
![1095919D-2561-4154-8E84-014C196C43F1](https://user-images.githubusercontent.com/115029650/193878304-e8a1bfcb-80d8-4d94-ad2d-ae13bf960ceb.jpeg)
![2F4D030D-0F14-4D9A-9BF0-DB0D205C0517](https://user-images.githubusercontent.com/115029650/193878322-a8553e9a-5733-4d0e-b539-d5ae8d6011ec.jpeg)
![732F6614-AD54-4072-967A-FCB83BDA3BE5](https://user-images.githubusercontent.com/115029650/193878336-ce8e0793-8a34-401c-8da0-9e29997bdbdb.jpeg)
![4FCFF5DC-834E-4942-9B61-E461C1BC6478](https://user-images.githubusercontent.com/115029650/193878349-f7544548-f01a-474f-b47e-ff79c64bb645.jpeg)
![0B857B07-36EA-428A-AC25-0CC41B149646](https://user-images.githubusercontent.com/115029650/193878361-1ee79716-5ded-4c18-a414-8e9732d40030.jpeg)

**Comment by SxeyChick** [comment]
Or you could have a mix of “Infill Support” and “Gradual Infill” for these common “island” areas: do gradual infill for every part of the print that goes bottom to top without dipping back down at all, and do infill support for the islands.

**Comment by SxeyChick** [comment]
If you’re wondering, this is printed with a 1.2 nozzle at .9 wall width, 2 walls. With that thickness of wall, I don’t need much infill for it to be strong. I was going to go with zigzag or lines, both of which have enough to give the object some weight and structure. But the embossed word would fully fail with any of these infill.

**Comment by canz** [comment]
i believe adding something like 'Bridge Sparse Infill Max Density' called 'Gradual Infill Sparse Max Density' and a corresponding infill speed\fan speed  would fix the issue for gradual Infill patterns trying to span a gap to raise the density of the infill
